src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -21,8 +21,12 @@ class EntityHookContext(
         return compService.holder(type)[this]
     }
 
-    inline fun <reified T : Component<*>> Entity.contains(componentType: ComponentType<T>): Boolean {
-        return compService.holder(componentType).contains(this)
+    inline fun <reified T : Component<*>> Entity.contains(type: ComponentType<T>): Boolean {
+        return compService.holder(type).contains(this)
+    }
+
+    inline fun <reified T : Component<*>> Entity.getOrNull(type: ComponentType<T>): T? {
```

Comment:
`contains` and `getOrNull` sound reasonable. We also need to add them to `IteratingSystem` and `Family`. I think we could avoid duplicating those methods by making the system and family also a `EntityHookContext` and delegate it to the world's hook instance but as far as I know this simply implements the methods in those classes by then calling the delegation instance (=hook instance of the world). So it is one additional function call and that's why I hesitate to use that approach.

But duplicating those "basic" entity extension functions in different places is also somehow annoying :D

Reply:
Yes, I also saw that both functions would be also useful in the system code. I already added them locally but then hadn't time to go on ;)
Yes, it looks annoying to have these functions spread around the codebase but they are luckily small and I also have no better idea :)